### PR TITLE
Simplify context in user space (part 1)

### PR DESCRIPTION
**Pull request by pj1031999**
- Unify `struct sigcontext` with `ucontext_t`.
- Remove regnum.h.
- Fix restoring of FPU context for _mips_.
- Remove machine/signal.h.

### CODE REVIEW

File: include/mips/signal.h [hunk 1]
```diff
@@ -55,17 +56,11 @@ typedef int sig_atomic_t;
  * a non-standard exit is performed.
  */
 #if defined(_LIBC) || defined(_KERNEL)
+
 struct sigcontext {
```
Comment (cahirwpz) [inline include/mips/signal.h:60]: Why can't we completely remove `sigcontext`? It looks artificial to have it defined like that.

File: lib/libc/gen/mips/longjmp.c [hunk 2]
```diff
@@ -68,28 +67,29 @@ void longjmp(jmp_buf env, int val) {
...
-  uc.uc_mcontext.__gregs[_REG_S0] = sc->sc_regs[_R_S0];
-  uc.uc_mcontext.__gregs[_REG_S1] = sc->sc_regs[_R_S1];
-  uc.uc_mcontext.__gregs[_REG_S2] = sc->sc_regs[_R_S2];
-  uc.uc_mcontext.__gregs[_REG_S3] = sc->sc_regs[_R_S3];
-  uc.uc_mcontext.__gregs[_REG_S4] = sc->sc_regs[_R_S4];
-  uc.uc_mcontext.__gregs[_REG_S5] = sc->sc_regs[_R_S5];
-  uc.uc_mcontext.__gregs[_REG_S6] = sc->sc_regs[_R_S6];
-  uc.uc_mcontext.__gregs[_REG_S7] = sc->sc_regs[_R_S7];
-  uc.uc_mcontext.__gregs[_REG_S8] = sc->sc_regs[_R_S8];
+  uc.uc_mcontext.__gregs[_REG_S0] = sc_uc->uc_mcontext.__gregs[_REG_S0];
```
Comment (cahirwpz) [inline lib/libc/gen/mips/longjmp.c:70]: Can we define nice macro to hide nicely access to `__gregs`?

File: lib/libc/gen/mips/longjmp.c [hunk 3]
```diff
@@ -68,28 +67,29 @@ void longjmp(jmp_buf env, int val) {
...
   /* Copy FP state */
-  if (sc->sc_fpused) {
+  if (sc_uc->uc_flags & _UC_FPU) {
     /* FP saved regs are $f20 .. $f31 */
-    memcpy(&uc.uc_mcontext.__fpregs.__fp_r.__fp_regs[20], &sc->sc_fpregs[20],
-           32 - 20);
-    uc.uc_mcontext.__fpregs.__fp_csr = sc->sc_fpregs[_R_FSR - _FPBASE];
+    memcpy(&uc.uc_mcontext.__fpregs.__fp_r.__fp_regs[20],
+           &sc_uc->uc_mcontext.__fpregs.__fp_r.__fp_regs[20],
+           (32 - 20) * sizeof(float));
```
Comment (cahirwpz) [inline lib/libc/gen/mips/longjmp.c:91]: I think it's better to use `sizeof(fpregister_t)` here.

File: lib/libc/gen/mips/setjmp.S [hunk 4]
```diff
@@ -66,11 +67,16 @@ NON_LEAF(setjmp, SETJMP_FRAME_SIZE, ra)
 	PTR_ADDU	a1, sp, CALLFRAME_SIZ		# pointer to stack_t
 	jal		_C_LABEL(sigaltstack)
 
-	move		a0, s0				# restore jmpbuf
+    move        a0, s0
```
Comment (cahirwpz) [inline lib/libc/gen/mips/setjmp.S:70]: Please configure your `vim` correctly for assembly files. Indentation should be 8 spaces.

**Review by cahirwpz — COMMENTED** [review]
Please clean it up a bit.

File: lib/libc/gen/mips/longjmp.c (hunk 2 above)
Comment (pj1031999) [reply lib/libc/gen/mips/longjmp.c]: done

File: lib/libc/gen/mips/longjmp.c (hunk 3 above)
Comment (pj1031999) [reply lib/libc/gen/mips/longjmp.c]: done

File: include/mips/signal.h (hunk 1 above)
Comment (pj1031999) [reply include/mips/signal.h]: We can :)

File: lib/libc/gen/mips/setjmp.S (hunk 4 above)
Comment (pj1031999) [reply lib/libc/gen/mips/setjmp.S]: done

File: lib/libc/gen/mips/_setjmp.S [hunk 5]
```diff
@@ -85,10 +82,7 @@ LEAF(_longjmp)
 	SAVE_GP(CALLFRAME_GP)
 
 	REG_PROLOGUE
-	REG_L		v0, _OFFSETOF_SC_REGS(a0)	# get magic number
 	REG_L		ra, _OFFSETOF_SC_PC(a0)
-	REG_LI		t0, 0xACEDBADE
-	bne		v0, t0, botch		# jump if error
```
Comment (j-piecuch) [inline lib/libc/gen/mips/_setjmp.S:91]: Might as well remove the code at the `botch` label.

Comment (pj1031999) [reply lib/libc/gen/mips/_setjmp.S:91]: done

**Pull request merged**